crates/wasi/src/preview2/host/udp.rs

```diff
@@ -129,6 +131,8 @@ impl<T: WasiView> udp::HostUdpSocket for T {
         if let Some(connect_addr) = remote_address {
             util::validate_remote_address(&connect_addr)?;
             util::validate_address_family(&connect_addr, &socket.family)?;
+            // We don't actually use the connecter, we just use it to verify that `connect_addr` is allowed
+            let _ = socket.pool.udp_connecter(connect_addr)?;
```

Comment:
Might not be the prettiest solution, but: could we use this same workaround to validate the remote address in unconnected outgoing datagram streams?